pkg/ucp/datamodel/resourceprovidersummary.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package datamodel
+
+import (
+	"fmt"
+
+	v1 "github.com/radius-project/radius/pkg/armrpc/api/v1"
+	"github.com/radius-project/radius/pkg/ucp/resources"
+)
+
+const (
+	// ResourceProviderResourceSummaryType is the resource type for the summary of a resource provider.
+	//
+	// These are **READONLY** virtual resources served from URLs like:
+	//
+	// /planes/radius/local/providers/
+	// /planes/radius/local/providers/Contoso.Example
+	ResourceProviderSummaryResourceType = "System.Resources/resourceProviderSummaries"
+)
+
+// ResourceProviderSummaryIDFromParts returns the resource id for a resource provider summary.
+//
+// Since ResourceProviderSummary is a virtual resource, the resource id is different from the URL used to access it.
+func ResourceProviderSummaryIDFromParts(scope string, name string) (resources.ID, error) {
```

Comment:
Discussed in private review. TLDR here:

- The summary is a pre-computed data structure based on the combination of values for many other types.
- We need somewhere to store it.
- It doesn't have a resource type or resource id because it's not *exactly* a resource.
- We make up a *virtual* resource so that it has a resource type and id.
- We follow this pattern in a few other places in UCP already.